doc/releases/release-notes-3.7.rst

```diff
@@ -211,9 +211,12 @@ Boards & SoC Support
 
 * Added support for these ARM boards:
```

Comment:
Just realized we might want to list the boards "diff" differently now, e.g. by vendor, since some boards are multi-arch. But that's for another PR :) cc @aescolar @nashif